mne_bids/write.py

```diff
@@ -690,12 +691,31 @@ def _write_raw_brainvision(raw, bids_fname, events):
     meas_date = raw.info['meas_date']
     if meas_date is not None:
         meas_date = _stamp_to_dt(meas_date)
-    write_brainvision(data=raw.get_data(), sfreq=raw.info['sfreq'],
+
+    # Try to preserve original dtype, if possible.
+    if raw.orig_format == 'single':
+        fmt = 'binary_float32'
+    elif raw.orig_format == 'short':
+        fmt = 'binary_int16'
+    else:
+        warn(f'Encountered data in "{raw.orig_format}" format, which is not '
+             f'supported by the BrainVision format. Converting to float32.',
+             RuntimeWarning)
+        fmt = 'binary_float32'
+
+    # writing with a resolution 1e-9 should be precise enough to avoid loss of
```

Comment:
Let's:

- add this suggestion
- check that CIs pass
- merge
- fix pybv https://github.com/bids-standard/pybv/issues/45
- make a new PR to mne-bids to remove the XXX suggestion from here

```suggestion
        fmt = 'binary_float32'

    # XXX: for now, always write to 'binary_float32' until issue with pybv is solved.
    # https://github.com/bids-standard/pybv/issues/45
    fmt = 'binary_float32'

    # writing with a resolution 1e-9 should be precise enough to avoid loss of
```

Reply:
Not so fast, I'm already working on updated behavior and I'm currently checking the tests ;)